lightning-persister/src/lib.rs

```diff
@@ -117,6 +119,27 @@ impl FilesystemPersister {
 		util::write_to_file(path, "network_graph".to_string(), network_graph)
 	}
 
+	/// Write the provided scorer to the path provided at `FilesystemPersister`
+	/// initialization, within a file called "scorer"
+	pub fn persist_scorer<'a, S: Deref>(data_dir: String, scorer: &'a S) -> Result<(), std::io::Error>
+	where
+		S::Target: WriteableScore<'a>
+	{
+		let path = PathBuf::from(data_dir);
+		fs::create_dir_all(path.clone())?;
+		let filename_with_path = get_full_filepath(path.clone(), "scorer".to_string());
+		let tmp_filename = format!("{}.tmp", filename_with_path.clone());
+
+		let file = fs::File::create(&tmp_filename)?;
+		let write_res = scorer.write(&mut BufWriter::new(file));
+		if let Err(e) = write_res.and_then(|_| fs::rename(&tmp_filename, filename_with_path)) {
+			let _ = fs::remove_file(&tmp_filename);
+			Err(e)
+		} else {
+			Ok(())
+		}
+	}
+
```

Comment:
This is a placeholder implementation I borrowed from combining ideas from the sample and #1404. I will change it after that PR is merged. 

I reckon it is probably best to use `utils::write_to_file`, but that currently requires D to be `DiskWriteable`, which may conflict with our new trait `WriteableScore`. Any advice?

Reply:
You can implement `DiskWriteable` for any `WriteableScore` to fix it. `DiskWriteable` is also removed in #1417 so you can also wait for that to go in first (but I'll let you and @johncantrell97 figure out which order y'all want to land these).

Reply:
@johncantrell97 got it, thanks for clarifying!